pkg/integrations/v4/dm/emitter_wrapper_with_ff_test.go

```diff
@@ -0,0 +1,65 @@
+package dm
+
+import (
+	"github.com/newrelic/infrastructure-agent/internal/agent/cmdchannel/fflag"
+	"github.com/newrelic/infrastructure-agent/internal/feature_flags"
+	"github.com/newrelic/infrastructure-agent/internal/integrations/v4/integration"
+	"github.com/newrelic/infrastructure-agent/pkg/fwrequest"
+	"github.com/newrelic/infrastructure-agent/pkg/integrations/v4/protocol"
+	"gotest.tools/assert"
+	"testing"
+)
+
+type fakeEmitter struct {
+	wasCalled bool
+}
+
+func (e *fakeEmitter) Send(req fwrequest.FwRequest) {
+	e.wasCalled = true
+}
+
+func TestEmitterWithFF_Send(t *testing.T) {
+
+	tests := []struct {
+		name                  string
+		initialFeatureFlags   map[string]bool
+		emitterWithRegister   bool
+		emitterWithNoRegister bool
+	}{
+		{
+			name:                  "register enabled",
+			initialFeatureFlags:   map[string]bool{fflag.FlagDMRegisterEnable: true},
+			emitterWithRegister:   true,
+			emitterWithNoRegister: false,
+		},
+		{
+			name:                  "register disabled",
+			initialFeatureFlags:   map[string]bool{fflag.FlagDMRegisterEnable: false},
+			emitterWithRegister:   false,
+			emitterWithNoRegister: true,
+		},
+		{
+			name:                  "register not set",
+			initialFeatureFlags:   map[string]bool{},
+			emitterWithRegister:   false,
+			emitterWithNoRegister: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			emitterWithRegister := &fakeEmitter{wasCalled: false}
+			emitterWithNoRegister := &fakeEmitter{wasCalled: false}
```

Comment:
there's no need to set up falses, bool vars empty (initialisation) value is false by default

Reply:
you are right, I will clean it